hi-ml-cpath/src/health_cpath/utils/callbacks.py

```diff
@@ -141,48 +142,43 @@ def rank_folder(self) -> Path:
         return self.outputs_folder / "loss_ranks"
 
     @property
-    def exception_folder(self) -> Path:
-        return self.outputs_folder / "loss_exceptions"
-
-    def zfill_epoch(self, epoch: int) -> str:
-        return str(epoch).zfill(len(str(self.max_epochs)))
+    def anomalies_folder(self) -> Path:
+        return self.outputs_folder / "loss_anomalies"
 
     def create_outputs_folders(self) -> None:
         folders = [
             self.cache_folder,
             self.scatter_folder,
             self.heatmap_folder,
             self.rank_folder,
-            self.exception_folder,
+            self.anomalies_folder,
         ]
         for folder in folders:
             os.makedirs(folder, exist_ok=True)
 
-    def reset_loss_cache(self) -> LossDictType:
+    def reset_loss_cache(self) -> LossCacheDictType:
         keys = [ResultsKey.LOSS, ResultsKey.SLIDE_ID]
         if self.save_tile_ids:
             keys.append(ResultsKey.TILE_ID)
         return {key: [] for key in keys}
 
-    def is_time_to_cache_loss_values(self, current_epoch: int) -> bool:
+    def get_filename(self, root_folder: Path, filename: str, epoch: int, order: Optional[str] = None) -> Path:
```

Comment:
Using `filename` in the name a method that returns a full path can be confusing, as that usually refers to the last part of a path.